File: leetcode/lc-us/cpp/2509.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    vector<int> cycleLengthQueries(int n, vector<vector<int>> &queries) {
        int m = (int) queries.size();
        vector<int> res(m);
        for (int i = 0; i < m; i++) {
            int x = queries[i][0], y = queries[i][1];
            int cnt = 1;
            while (x != y) {
                if (x > y) x >>= 1;
                else y >>= 1;
                cnt++;
            }
            res[i] = cnt;
        }
        return res;
    }
};

File: leetcode/lc-us/py/2509.py
from typing import List


class Solution:
    def cycleLengthQueries(self, n: int, queries: List[List[int]]) -> List[int]:
        res = [0] * len(queries)
        for i in range(len(queries)):
            x, y = queries[i]
            t = 1
            while x != y:
                if x > y:
                    x >>= 1
                else:
                    y >>= 1
                t += 1
            res[i] = t
        return res
